### ISSUE TITLE
Counting number of transactions is slow on big addresses

### ISSUE THREAD

**Issue by tdroxler** [alephium/explorer-backend#459]
Some big players like https://explorer.alephium.org/addresses/1DKWCEc9HjhchKoMCMA4J1nbNGNjrAoPFnkcgA1w6h7jh and  https://explorer.alephium.org/addresses/1Gzp3Wc42fanrbAqknYmSaZLfC567bP3JWprDCiuJ8rbP are slow to get the number of transactions. 
PG is know for being slow at counting, we have several way to do it, [one is to estimate the number of count](https://www.cybertec-postgresql.com/en/postgresql-count-made-fast/), but unfortunately it also estimate small address.

Another solution is to count but with a limit

```sql
SELECT count(*) 
FROM (
  SELECT 1 
  FROM transaction_per_addresses 
  WHERE address  = '1Gzp3Wc42fanrbAqknYmSaZLfC567bP3JWprDCiuJ8rbP' 
  LIMIT 50000
) t;
```

Like this we could quickly count the small addresses, but warn the user if it goes beyond the limit.

This would impact the UI/UX and the pagination, so we need to discuss this @mvaivre @nop33

**Comment by tdroxler** [comment]
We could also think of introducing our [Hot addresses service](https://github.com/alephium/explorer-backend/issues/385)

**Comment by nop33** [comment]
The `txNumber` which comes from the `GET:/addresses/{address}` endpoint is used to:
1. Explorer:
   1. Nb. of transactions in Address page header
   1. Whether to show the "Download CSV" button or not
   1. Calculate end of pagination tx table in Address page
1. Desktop wallet:
   1. Determine whether to show the "Show more" button in the bottom of a tx table by comparing
      ```ts
      confirmedTxs.length !== totalNumberOfTransactions
      ``` 

I would say that for 1.1, 1.2, and 2.1, there's no problem on our side showing something like `> 100'000` instead of the actual number and adjusting the code to work appropriately. The only tricky part might be 1.3, but we'll figure it out.

What would be nice is if we could know whether the `txNumber` is an estimation or not so that we can adapt the frontend side accordingly.

**Comment by tdroxler** [comment]
yes of course we would add something like a boolean field `estimation: true` 
Maybe we could also have a query param on the endpoint to remove the estimation, if someone want the real number.

**Comment by tdroxler** [comment]
well this address with > 13M addresses seems to be fast and precise https://etherscan.io/address/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2

**Comment by polarker** [comment]
It's not precise for USDT: https://etherscan.io/token/0xdac17f958d2ee523a2206206994597c13d831ec7
The result is not updated that often, so they return `more than X transactions`.

**Comment by tdroxler** [comment]
We could optimize transaction counting for large addresses by maintaining a summary table:

```sql
CREATE TABLE address_transaction_counts (
  address TEXT PRIMARY KEY,
  tx_count BIGINT,
  at TIMESTAMP
);
```

The idea is to leverage the `FinalizerService` to periodically update `tx_count` with the total number of transactions for each address up to the latest finalized block (`at`). 

**How this would work:**
- When querying the transaction count for an address, we first read `tx_count` and `at` from this table.
- For transactions after `at`, we count only those newer records in `transaction_per_addresses`—i.e., `WHERE timestamp > at`.
- The final result is the sum of the cached count (`tx_count`) and the count of new transactions since the last update.

**Advantages:**
- For the majority of queries, especially for addresses with many transactions, this will be extremely fast.
- Only a small subset of recent transactions need to be counted on each query, reducing load and latency.
- The table can be updated asynchronously by the `FinalizerService`, keeping the counts accurate and up-to-date.

**Potential Implementation Steps:**
1. Update `address_transaction_counts` as part of the finalization process for each finalized block.
2. On transaction count queries, fetch `tx_count` and `at`, then count only new transactions after `at`.
3. Periodically clean up or backfill the table as needed.

This approach provides a scalable solution with minimal impact on live query performance, and should greatly improve responsiveness for large addresses.

**Comment by tdroxler** [comment]
It could actually be done as the same time as https://github.com/alephium/explorer-backend/issues/594 as it's the same approach
